src/libsync/abstractpropagateremotedeleteencrypted.cpp

```diff
@@ -0,0 +1,200 @@
+    : QObject(parent)
+    , _propagator(propagator)
+    , _item(item)
+{}
+
+QNetworkReply::NetworkError AbstractPropagateRemoteDeleteEncrypted::networkError() const
+{
+    return _networkError;
+}
+
+QString AbstractPropagateRemoteDeleteEncrypted::errorString() const
+{
+    return _errorString;
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::storeFirstError(QNetworkReply::NetworkError err)
+{
+    if (_networkError == QNetworkReply::NetworkError::NoError) {
+        _networkError = err;
+    }
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::storeFirstErrorString(const QString &errString)
+{
+    if (_errorString.isEmpty()) {
+        _errorString = errString;
+    }
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::startLsColJob(const QString &path)
+{
+    qCDebug(ABSTRACT_PROPAGATE_REMOVE_ENCRYPTED) << "Folder is encrypted, let's get the Id from it.";
+    auto job = new LsColJob(_propagator->account(), path, this);
+    job->setProperties({"resourcetype", "http://owncloud.org/ns:fileid"});
+    connect(job, &LsColJob::directoryListingSubfolders, this, &AbstractPropagateRemoteDeleteEncrypted::slotFolderEncryptedIdReceived);
+    connect(job, &LsColJob::finishedWithError, this, &AbstractPropagateRemoteDeleteEncrypted::taskFailed);
+    job->start();
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::slotFolderEncryptedIdReceived(const QStringList &list)
+{
+    qCDebug(ABSTRACT_PROPAGATE_REMOVE_ENCRYPTED) << "Received id of folder, trying to lock it so we can prepare the metadata";
+    auto job = qobject_cast<LsColJob *>(sender());
+    const ExtraFolderInfo folderInfo = job->_folderInfos.value(list.first());
+    slotTryLock(folderInfo.fileId);
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::slotTryLock(const QByteArray &folderId)
+{
+    auto lockJob = new LockEncryptFolderApiJob(_propagator->account(), folderId, this);
+    connect(lockJob, &LockEncryptFolderApiJob::success, this, &AbstractPropagateRemoteDeleteEncrypted::slotFolderLockedSuccessfully);
+    connect(lockJob, &LockEncryptFolderApiJob::error, this, &AbstractPropagateRemoteDeleteEncrypted::taskFailed);
+    lockJob->start();
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::slotFolderLockedSuccessfully(const QByteArray &folderId, const QByteArray &token)
+{
+    qCDebug(ABSTRACT_PROPAGATE_REMOVE_ENCRYPTED) << "Folder id" << folderId << "Locked Successfully for Upload, Fetching Metadata";
+    _folderLocked = true;
+    _folderToken = token;
+    _folderId = folderId;
+
+    auto job = new GetMetadataApiJob(_propagator->account(), _folderId);
+    connect(job, &GetMetadataApiJob::jsonReceived, this, &AbstractPropagateRemoteDeleteEncrypted::slotFolderEncryptedMetadataReceived);
+    connect(job, &GetMetadataApiJob::error, this, &AbstractPropagateRemoteDeleteEncrypted::taskFailed);
+    job->start();
+}
+
+void AbstractPropagateRemoteDeleteEncrypted::slotFolderUnLockedSuccessfully(const QByteArray &folderId) {
```

Comment:
Sorry didn't spot it before apparently, opening curly brace on its own line please. ;-)